### ISSUE TITLE
A request has failed from JSON Language Features

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#86140]
As of today whenever I open a new window I get this message:

<img width="464" alt="Screen Shot 2019-12-03 at 10 55 16 AM" src="https://user-images.githubusercontent.com/2193314/70080486-7ee18d00-15bb-11ea-8738-6290ab22e7f1.png">
<img width="262" alt="Screen Shot 2019-12-03 at 10 55 21 AM" src="https://user-images.githubusercontent.com/2193314/70080489-8012ba00-15bb-11ea-96c1-cb48695a6f32.png">

The console is complaining about the lack of a corrupted.info file?

<img width="1082" alt="Screen Shot 2019-12-03 at 10 54 45 AM" src="https://user-images.githubusercontent.com/2193314/70080510-87d25e80-15bb-11ea-820f-d21bc8220d34.png">

It appears related to the en-gb language pack that I have installed? https://marketplace.visualstudio.com/items?itemName=MS-CEINTL.vscode-language-pack-en-GB

Version: 1.41.0-insider
Commit: 97855786a014be2440751b038b373c3726e11fe8
Date: 2019-12-03T05:35:54.992Z
Electron: 6.1.5
Chrome: 76.0.3809.146
Node.js: 12.4.0
V8: 7.6.303.31-electron.0
OS: Darwin x64 19.0.0

**Comment by dbaeumer** [comment]
IMO the console complaining about the lack of a corrupted.info file is not the cause of the JSON language server to fail. In case this fails it will continue showing default English strings.

I create a new issue in the vscode-nls repository which actually prints that message. It is: https://github.com/microsoft/vscode-nls/issues/27

Assigning to @aeschli for the JSON language server problem.

**Comment by dbaeumer** [comment]
@Tyriar can you please comment on https://github.com/microsoft/vscode-nls/issues/27 so that I undertstand what could cause this.

**Comment by aeschli** [comment]
@Tyriar Can you enable `"json.trace.server": "verbose"` and check the `JSON Language Server` output?

**Comment by Tyriar** [comment]
Both the message and console errors are not happening anymore this morning, the clp directory has a different hash in the folder name now?

**Comment by dbaeumer** [comment]
Yes, since it is a new VS Code version. Does the path that failed still exists. I only clean them after a couple of weeks. If it doesn't it would explain the exception but I would still not understand why it wasn't there in the first place.

**Comment by aeschli** [comment]
Closing. Please reopen if you see it again.

**Comment by Tyriar** [comment]
I keep hitting this, it may be related to corrupted.info? Here's a snippet from the enormous log file after enabling verbose:

```
[Trace - 11:32:28 am] Received notification 'window/logMessage'.
Params: {
    "type": 1,
    "message": {
        "errno": -4058,
        "code": "ENOENT",
        "syscall": "open",
        "path": "C:\\Users\\Daniel\\AppData\\Roaming\\Code - Insiders\\clp\\918cf835f834df511eef0fd61f736424.en-gb\\corrupted.info"
    }
}


[Error - 11:32:28 am] [object Object]
```

**Comment by aeschli** [comment]
The JSON LS does not write to console, but it uses `vscode-nls`, which does.
I guess it's this line:
https://github.com/microsoft/vscode-nls/blob/fe0fbef1c9825767c40d70d8d7817c31b31cd3ab/src/main.ts#L185

**Comment by dbaeumer** [comment]
The `vscode-nls` tries to mark the caches language pack as corrupted but can't: it is here https://github.com/Microsoft/vscode-nls/blob/master/src/main.ts#L185 

@aeschli are you handling console.error special in our language server ?

@Tyriar can you always reproduce this with that language pack?

**Comment by aeschli** [comment]
The JSON LS does the following:
`console.error = connection.console.error.bind(connection.console);`

**Comment by dbaeumer** [comment]
That explains it. But it wouldn't be necessary for errors when running in ipc transport since stderr and stdout are capture correctly on the client side.

**Comment by dbaeumer** [comment]
@Tyriar I tried to reproduce this using teh en.gb LP but can't. Any additional steps ? 

I would like to understand why this is happening before I add some protection code.

**Comment by 7iomka** [comment]
the same issue when I open any json file

**Comment by Tyriar** [comment]
@dbaeumer there's no special steps, just opening settings.json for the first time after a reload.

**Comment by dbaeumer** [comment]
@Tyriar actually not for me :-(

![capture](https://user-images.githubusercontent.com/1931590/76837011-1056d100-6832-11ea-9016-dc7741cf5c69.png)

Anything you could think of that would make this reproducable.

@7iomka which locale are you using and on which OS are you seeing this?

**Comment by dbaeumer** [comment]
I puplished 4.1.2 of vscode-nls that adds some protection code against this to ensure that the diurectory still exists when trying to write the corrupted file.

**Comment by dbaeumer** [comment]
Upstream https://github.com/microsoft/vscode-nls/issues/27

**Comment by 7iomka** [comment]
> @7iomka which locale are you using and on which OS are you seeing this?

![image](https://user-images.githubusercontent.com/18473137/76861909-fd5df400-6865-11ea-9d16-7ac173b42d19.png)
Russian locale
Mac OS Catalina (latest)
Error happened if one of file with json syntax is opened on first run, or after first open json file after reload editor.
![image](https://user-images.githubusercontent.com/18473137/76862218-6e9da700-6866-11ea-9932-0708fde760fa.png)

**Comment by dbaeumer** [comment]
@7iomka does 

![capture](https://user-images.githubusercontent.com/1931590/76952615-a7448b80-690d-11ea-84ea-c31d9a9dd038.png)


this directory exist on your system (e.g. the directory containing the `corrupted.info`)

**Comment by dbaeumer** [comment]
To get rid of the error you could close VSCode and delete the clp directory.

**Comment by dbaeumer** [comment]
@aeschli why are you doing this: https://github.com/microsoft/vscode/issues/86140#issuecomment-596450387

If you use Node IPC as a transport console error and console out will be handled on the client and put into the client console. 

You could easily print console.error from other libraries as well

**Comment by aeschli** [comment]
@dbaeumer But what if stdout is used as transport? The JSON language server is published as npm module and there's currently no limitation on the transport.

**Comment by dbaeumer** [comment]
I will had some code to the library that doesn't show a message box in case the error comes in via a `conosle.*`

**Comment by dbaeumer** [comment]
Published client @ 6.1.2.
